PyDSS/pyPostprocessor/PostprocessScripts/EdLiFoControl.py

```diff
@@ -0,0 +1,698 @@
+        self.total_kW_output = 0
+        Pct_kW_curtailment = 0
+
+        for pv_sys in self.poa_lifo_ordered_pv_list:
+            # pv_sys_n = remove_pen(pv_sys)
+            self.PV_curtailed[pv_sys] = 0
+            self.pvbus_voltage[pv_sys] = []
+            # PV_kW_curtailed[pv_sys_n] = 0
+            self.PV_kW_curtailed[pv_sys + "NetCurt"] = 0
+            self.PV_kW_curtailed[pv_sys + "kWCurt"] = 0
+            self.PV_kW_curtailed[pv_sys + "PctCurt"] = 0
+            self.PV_kW_curtailed[pv_sys + "PctPmpp"] = 100
+
+        self.overloads_df, self.affected_buses = check_line_overloads(
+            self.monitored_lines
+        )
+        self.logger.info(f"Number of buses affected: {len(self.affected_buses)}")
+
+        self.zone = []
+
+        if self.affected_buses:
+
+            self.zone = form_pvzones(
+                self.affected_buses, self.dismat_df, self.zone_threshold
+            )
+            # print('zone length:',len(zone))
+
+        else:
+            self.logger.info("No monitored line affected")
+
+        self.get_curtailment_candidates()
+
+        self.solution_status = 0
+
+        if self.pvs_df.loc[self.poa_lifo_ordered_pv_list, "kVARated"].max() == 0:
+            self.comment = "No PV system found!"
+            self.logger.info(self.comment)
+            try:
+                number_ol = len(self.overloads_df)
+                self.logger.info(f"There are {number_ol} persisting overloads")
+
+            except:
+                pass
+
+        elif self.my_lifo_list:
+
+            self.init_pctpmpp = {}
+            self.init_kVA = {}
+            self.init_net_power_PV = 0
+            self.init_power_PV = []
+
+            self.solve_overloads()
+
+        if self.solution_status == -1:
+            self.comment = "No need to curtail!"
+
+        if max(self.PV_curtailed.keys(), key=(lambda k: self.PV_curtailed[k])) == 0:
+            self.comment = "Deployment passed without curtailment!"
+            # print(comment)
+            self.solution_status == 0
+
+        self.PV_curtailed["Comment"] = self.comment
+        Pct_curtailment = self.total_kVA_curtailed * 100 / max(1, total_kVA_deployed)
+        self.PV_curtailed["Pct_curtailment"] = Pct_curtailment
+
+        Pct_kW_curtailment = (
+            abs(self.total_kW_curtailed)
+            * 100
+            / max(1, abs(self.total_init_kW_deployed))
+        )
+        self.PV_kW_curtailed["Pct_kW_curtailment"] = Pct_kW_curtailment
+        self.PV_kW_curtailed["Total_PV_kW_deployed"] = self.total_init_kW_deployed
+        self.PV_kW_curtailed["Total_PV_kW_curtailed"] = self.total_kW_curtailed
+        self.PV_kW_curtailed["Total_PV_kW_output"] = self.total_kW_output
+
+        self.PV_kW_curtailed["Comment"] = self.comment
+        # print("Done with 1 poa loop:", Pct_curtailment)
+
+        self.logger.info(f"kW Curtailed: {self.PV_kW_curtailed}")
+        self.has_converged = sol.Converged()
+        self.error = (

```

Comment:
This seems weird...did `black` do it? The outer parens are not necessary.

Reply:
Yes